Firestore/Source/Remote/FSTDatastore.h

```diff
@@ -139,7 +139,9 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * See https://github.com/grpc/grpc/issues/10957 for the kinds of things we're trying to avoid.
  */
-@interface FSTStream : NSObject
+@interface FSTStream <__covariant FSTStreamDelegate> : NSObject
+
+@property (nonatomic, weak, readwrite, nullable) FSTStreamDelegate delegate;
```

Comment:
Does the `delegate` need to be publicly declared? Should anyone outside the class access it? If not, consider just moving this into the implementation.

If someone needs to read it you can declare it `readonly` here and redeclare it internally as `readwrite`. Note that conventionally we haven't been marking properties `readwrite` since that's the default.

Reply:
It doesn't need to be but it means that we have to redeclare it in the Mock implementations (unless we expose an test only interface)